File: php/sortedSquares.php
<?php

class Solution {

     // brute force O(n ^ 2)
    function bruteForce($nums) {
        for ($i=0; $i < count($nums); $i++) { 
            $nums[$i] = $nums[$i] * $nums[$i];

        }
        for ($i=0; $i < count($nums); $i++) { 
            for ($j=$i+1; $j < count($nums); $j++) { 
                if($nums[$i] > $nums[$j]) {
                    $temp= $nums[$i];
                    $nums[$i] =$nums[$j];
                    $nums[$j]=$temp;
                }
            }
        }

        return $nums;
    }

    function sortedSquares($nums) {
        for ($i=0; $i < count($nums); $i++) { 
            $nums[$i] = $nums[$i] * $nums[$i];

        }

        $arr = array_fill(0, count($nums), 0);
        $left = 0;
        $right = count($nums) - 1;

        for ($i=count($nums) - 1; $i >= 0 ; $i--) { 
           $leftNumber = $nums[$left];
           $rightNumber = $nums[$right];

           if($leftNumber < $rightNumber) {
            $arr[$i] = $rightNumber;
            $right--;
           }else {
            $arr[$i] = $leftNumber;
            $left++;
           }

        }

        return $arr;
    }
}

$nums = [-4,-1,0,3,10];

$solution = new Solution();

print_r($solution->sortedSquares($nums));

File: 2025/leetcode /sortedSquares.py
from typing import List

# class Solution:
#     def sortedSquares(self, nums: List[int]) -> List[int]:
#         index = 0
#         while index < len(nums):
#           num = nums[index]
#           nums[index] = num * num
#           index += 1

#         nums.sort()
#         return nums

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        index = right 

        result = [0] * len(nums)

        while index >= 0 :
          leftSquare = nums[left] * nums[left]
          rightSquare = nums[right] * nums[right]

          if leftSquare > rightSquare :
            result[index] = leftSquare
            left += 1
          else :
            result[index] = rightSquare
            right -= 1
      
          index -= 1

        return result

solution = Solution()

print(solution.sortedSquares([-7,-3,2,3,11]))